### PR TITLE
Readme edits

### PR DESCRIPTION
**Pull request by JohnTheCoolingFan**
Added `--release` flag to build instructions, which makes cargo build non-debug version with improved performance.

Added new API docs to readme.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,9 +32,13 @@ int mtzip_zip_archive_add_file_from_bytes(mtzip_zip_archive_t* zip_archive, uint
 
 int mtzip_zip_archive_add_directory(mtzip_zip_archive_t* zip_archive, const char* archive_name);
 
-int mtzip_zip_archive_compress(mtzip_zip_archive_t* zip_archive, size_t threads);
+int mtzip_zip_archive_compress(mtzip_zip_archive_t* zip_archive);
```
Comment (thechampagne) [inline README.md]: It should be `int mtzip_zip_archive_compress(mtzip_zip_archive_t* zip_archive, size_t threads);`

File: README.md [hunk 2]
```diff
@@ -32,9 +32,13 @@ int mtzip_zip_archive_add_file_from_bytes(mtzip_zip_archive_t* zip_archive, uint
...
 
-int mtzip_zip_archive_write(mtzip_zip_archive_t* zip_archive, const char* file_name, size_t threads);
+int mtzip_zip_archive_compress_autothread(mtzip_zip_archive_t* zip_archive, size_t threads);
```
Comment (thechampagne) [inline README.md]: It should be `int mtzip_zip_archive_compress_autothread(mtzip_zip_archive_t* zip_archive);`

File: README.md [hunk 3]
```diff
@@ -32,9 +32,13 @@ int mtzip_zip_archive_add_file_from_bytes(mtzip_zip_archive_t* zip_archive, uint
...
+
+int mtzip_zip_archive_write(mtzip_zip_archive_t* zip_archive, const char* file_name);
```
Comment (thechampagne) [inline README.md]: It should be `int mtzip_zip_archive_write(mtzip_zip_archive_t* zip_archive, const char* file_name, size_t threads);`

File: README.md [hunk 4]
```diff
@@ -32,9 +32,13 @@ int mtzip_zip_archive_add_file_from_bytes(mtzip_zip_archive_t* zip_archive, uint
...
+
+int mtzip_zip_archive_write_autothread(mtzip_zip_archive_t* zip_archive, const char* file_name, size_t threads);
```
Comment (thechampagne) [inline README.md]: It should be `int mtzip_zip_archive_write_autothread(mtzip_zip_archive_t* zip_archive, const char* file_name);`

**Review by thechampagne — CHANGES_REQUESTED** [review]

**Comment by JohnTheCoolingFan** [comment]
Ah, right, got them mixed up. Fixed!

**Pull request merged**